runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/SpecialCounterPlugin.java

```diff
@@ -0,0 +1,255 @@
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.VarbitChanged;
+import net.runelite.api.queries.InventoryItemQuery;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+import net.runelite.client.util.QueryRunner;
+
+import javax.inject.Inject;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+@PluginDescriptor(
+	name = "Special Attack Counter",
+	enabledByDefault = false
+)
+public class SpecialCounterPlugin extends Plugin
+{
+	private int currentWorld = -1;
+	private int specialPercentage = -1;
+	private int specialHitpointsExperience = -1;
+	private boolean specialUsed;
+	private double modifier = 1d;
+
+	private SpecialWeapon specialWeapon;
+	private String interactedNPC = null;
+	private final SpecialCounter[] specialCounter = new SpecialCounter[SpecialWeapon.values().length];
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private InfoBoxManager infoBoxManager;
+
+	@Inject
+	private ItemManager itemManager;
+
+	@Inject
+	private QueryRunner queryRunner;
+
+	@Inject
+	private ConfigManager configManager;
+
+	@Override
+	protected void shutDown()
+	{
+		removeCounter();
+	}
+
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged event)
+	{
+		if (event.getGameState() == GameState.LOGGED_IN)
+		{
+			if (currentWorld == -1)
+			{
+				currentWorld = client.getWorld();
+			}
+			else if (currentWorld != client.getWorld())
+			{
+				currentWorld = client.getWorld();
+				removeCounter();
+			}
+		}
+	}
+
+	@Subscribe
+	private void onSpecialChanged(VarbitChanged event)
+	{
+		int specialPercentage = client.getSetting(Setting.SPECIAL_ATTACK_PERCENT);
+
+		if (this.specialPercentage == -1 || specialPercentage >= this.specialPercentage)
+		{
+			this.specialPercentage = specialPercentage;
+			return;
+		}
+		else
+		{
+			this.specialPercentage = specialPercentage;
+			this.specialWeapon = usedSpecialWeapon();
+
+			Player localPlayer = client.getLocalPlayer();
+
+			if (localPlayer.getInteracting() != null && !localPlayer.getInteracting().getName().equals(interactedNPC))
+			{
+				interactedNPC = localPlayer.getInteracting().getName();
+
+				if (Boss.getBoss(interactedNPC) != null)
+				{
+					modifier = Boss.getBoss(interactedNPC).getModifier();
+				}
+				else
+				{
+					modifier = 1d;
+				}
+			}
+
+			specialUsed = true;
+			specialHitpointsExperience = client.getSkillExperience(Skill.HITPOINTS);
+		}
+	}
+
+	@Subscribe
+	private void onGameTick(GameTick tick)
+	{
+		Player localPlayer = client.getLocalPlayer();
+
+		if (localPlayer.getInteracting() != null && !localPlayer.getInteracting().getName().equals(interactedNPC))
+		{
+			interactedNPC = localPlayer.getInteracting().getName();
+			removeCounter();
+		}
+
+		if (specialHitpointsExperience != -1 && specialUsed)
```

Comment:
There is an event for xp change, does that work here? Or is it prior to the special bar precent dropping?

Reply:
Initially I tried using the experience change event but I couldn't sync it properly with the special drain to check for a delta in experience, so I came up with the current solution.